src/app/clusters/ota-requestor/DefaultOTARequestor.cpp

```diff
@@ -522,22 +522,42 @@ void DefaultOTARequestor::TriggerImmediateQueryInternal()
     ConnectToProvider(kQueryImage);
 }
 
-OTARequestorInterface::OTATriggerResult DefaultOTARequestor::TriggerImmediateQuery()
+CHIP_ERROR DefaultOTARequestor::TriggerImmediateQuery(FabricIndex fabricIndex)
 {
     ProviderLocationType providerLocation;
-    bool listExhausted = false;
-    if (mOtaRequestorDriver->GetNextProviderLocation(providerLocation, listExhausted) != true)
+    bool providerFound = false;
+
+    if (fabricIndex == kUndefinedFabricIndex)
+    {
+        bool listExhausted = false;
+        providerFound      = mOtaRequestorDriver->GetNextProviderLocation(providerLocation, listExhausted);
+    }
+    else
+    {
+        for (auto providerIter = mDefaultOtaProviderList.Begin(); providerIter.Next();)
+        {
+            providerLocation = providerIter.GetValue();
+
+            if (providerLocation.GetFabricIndex() == fabricIndex)
+            {
+                providerFound = true;
+                break;
+            }
+        }
+    }
+
+    if (!providerFound)
     {
         ChipLogError(SoftwareUpdate, "No OTA Providers available");
-        return kNoProviderKnown;
+        return CHIP_ERROR_INCORRECT_STATE;
     }
 
     SetCurrentProviderLocation(providerLocation);
 
     // Go through the driver as it has additional logic to execute
     mOtaRequestorDriver->SendQueryImage();
```

Comment:
Suggest adding a log about triggering immeidate OTA query for fabric Index 0x%x with providerLocation.GetFabricIndex()